internal/environment/envpool.go

```diff
@@ -0,0 +1,19 @@
+package environment
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aau-network-security/haaukins-agent/internal/environment/lab"
+)
+
+func (ep *EnvPool) GetLabByTag(tag string) (*lab.Lab, error) {
+	for ke, e := range ep.Envs {
+		for kl, l := range e.Labs {
+			if l.Tag == tag {
+				return ep.Envs[ke].Labs[kl], nil
+			}
+		}
+	}
```

Comment:
I think we can lower the complexity of this function but ill need. a bit more time to look at the source :)